src/ripple/app/main/Application.cpp

```diff
@@ -1289,156 +1299,198 @@ ApplicationImp::getLastFullLedger()
     }
 }
 
-bool ApplicationImp::loadOldLedger (
-    std::string const& ledgerID, bool replay, bool isFileName)
+std::shared_ptr<Ledger>
+ApplicationImp::loadLedgerFromFile (
+    std::string const& name)
 {
     try
     {
-        std::shared_ptr<Ledger> loadLedger, replayLedger;
+        std::ifstream ledgerFile (name, std::ios::in);
 
-        if (isFileName)
+        if (!ledgerFile)
+        {
+            JLOG(m_journal.fatal()) <<
+                "Unable to open file '" << name << "'";
+            return nullptr;
+        }
+
+        Json::Reader reader;
+        Json::Value jLedger;
+
+        if (!reader.parse (ledgerFile, jLedger))
         {
-            std::ifstream ledgerFile (ledgerID.c_str (), std::ios::in);
-            if (!ledgerFile)
+            JLOG(m_journal.fatal()) <<
+                "Unable to parse ledger JSON";
```

Comment:
Might it be useful if this said `"Unable to parse ledger " << name << " JSON."`?